src/main/java/org/spdx/sbom/gradle/utils/SpdxDocumentBuilder.java

```diff
@@ -272,20 +271,20 @@ private Optional<SpdxPackage> createMavenModulePackage(
         throw new RuntimeException("Source repo was null?");
       }
 
-      var baseURI = mavenArtifactRepositories.get(sourceRepo);
-      String modulePath =
-          moduleId.getGroup().replace(".", "/")
-              + "/"
-              + moduleId.getName()
-              + "/"
-              + moduleId.getVersion()
-              + "/"
-              + URLEncoder.encode(dependencyFile.getName(), StandardCharsets.UTF_8);
-      URI downloadLocation = baseURI.resolve(modulePath);
+      var repoUri = mavenArtifactRepositories.get(sourceRepo);
       if (taskExtension != null) {
-        downloadLocation = taskExtension.mapDownloadUri(downloadLocation);
+        repoUri = taskExtension.mapRepoUri(repoUri);
```

Comment:
Oops, I definitely said that it sounded good to me for mapRepoUri to just take as input the repository parameter, but I think I was wrong, since in AndroidX we could have one repository that contains the remote versions of artifacts from several other artifacts

Fortunately I think there's an easy fix: could we also pass moduleId into mapRepoUri?